pinotdb/db.py

```diff
@@ -597,17 +597,22 @@ async def close(self):
 
 def apply_parameters(operation, parameters):
     escaped_parameters = {
-        key: escape(value) for key, value in parameters.items()}
-    return operation % escaped_parameters
+        key: escape_parameter(value) for key, value in parameters.items()}
+    escaped_operation = escape_operation(operation)
+    return escaped_operation % escaped_parameters
 
 
-def escape(value: Any) -> Any:
+def escape_parameter(value: Any) -> Any:
     if value == "*":
         return value
     elif isinstance(value, str):
         return "'{}'".format(value.replace("'", "''"))
     elif isinstance(value, bool):
         return "TRUE" if value else "FALSE"
     elif isinstance(value, (list, tuple)):
-        return ", ".join(str(escape(element)) for element in value)
+        return ", ".join(str(escape_parameter(element)) for element in value)
     return value
+
+
+def escape_operation(value: str) -> str:
+    return value.replace('%', '%%').replace('%(', '(')
```

Comment:
One concern I have is what if users are already handling this escaping while sending the query. Should we handle that with `return value.replace('%%', '%')replace('%', '%%').replace('%(', '(')` ?

Reply:
in that case maybe you can do 
```
value.replace('%%', '%').replace('%', '%%').replace('%(', '(')
```

Reply:
Or make it a place holder, then convert it back, e.g. 
```
import re

def convert_percent_correctly(text):
    # First, temporarily replace '%%' with a placeholder to avoid modifying them
    temp_placeholder = "TEMP_DOUBLE_PERCENT"
    text = text.replace('%%', temp_placeholder)
    
    # Then, convert standalone '%' to '%%'
    text = re.sub(r'%(?!\()', '%%', text)
    
    # Finally, replace the placeholder back to '%%'
    text = text.replace(temp_placeholder, '%%')
    
    return text
```

Reply:
That's much cleaner!